orangecontrib/educational/widgets/owkmeans.py

```diff
@@ -472,13 +456,15 @@ def number_of_clusters_change(self):
         """
         if self.numberOfClusters > len(self.data):
             # if too less data for clusters number
-            self.warning(2, "Please provide at least number of points equal to "
-                            "number of clusters selected or decrease number of clusters")
+            self.warning(
+                2, "The number of clusters can't exceed the number of points")
```

Comment:
Aim for shorter messages. Don't use "please" in these messages. I guess it works better if you state the fact and let the user figure it out. :)